### PR TITLE
Part I - Yet another introduction to concurrency

### PR DESCRIPTION
**Pull request by nandosousafr**
Feel free to review it ;)

### CODE REVIEW

File: _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+When the industry took this way, not only _hardware production_ was affected,
+but also the way we write software. For several decades, Software Engineers
+enjoyed free performance gains due to CPU optimizations predicted by Moore's
+law.  However, today if you want fully exploit your machine, you need to
+exploit __parallelism__.
+
+### Concurrency vs Parallelism
+
+You may say:
```
Comment (nandosousafr) [inline _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md]: you may ask? would be better?

File: _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+I've started to professionally work with programming in 2011. One year later, I
+have discovered the Ruby language and have fallen in love with it. Since then
+I've worked on a plenty of projects and have never worried about concurrency.
+Meanwhile, there have been some discussions on how Ruby is inefficient in terms
+of concurrency. This instigated me to study the topic to really understand what
+was happening.
+
+After sometime studying it, I noticed that some of
+developers (even seniors), had the same lack of knowledge as I had. For this
```
Comment (LeoSL) [inline _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md]: "That's the reason why I'm writing this article: an attempt to help other developers with the same gap"

File: _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+reason, I'm writing  this article as an attempt to help you if you have this
+same gap.
+
+## Why is everyone talking about it?
+
```
Comment (LeoSL) [inline _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md]: Why old languages, like Haskell and Erlang**,** are hitting trending topics?

File: _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+when Gordon Moore made an observation:
+
+>  The number of transistors per square inch on integrated circuits had doubled
+>  every year since the integrated circuit was invented.
+
+
+But what it means? In an updated version, it means: _"The machine processing
+power tends to double every 18 months"_. However, we're experiencing a slow down
+in the pace of this observation due to problems like: heat leakage and energy
+consumption. So, Instead of shipping doubled faster chips every 18 months, the
```
Comment (LeoSL) [inline _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md]: Instead ficou maísculo.

File: _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+industry took another direction: _Multicores_.
+
+### Multicore
+
+When the industry took this way, not only hardware production was affected, but
```
Comment (LeoSL) [inline _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md]: ...the industry took this path

File: _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+also the way we write software. For several decades, Software Engineers enjoyed
+free performance gains. However, today if you want fully exploit your machine,
```
Comment (LeoSL) [inline _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md]: if you want **to** ...

File: _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+### Concurrency vs Parallelism
+
+You may ask:
+> Parallelism? you're talking about concurrency, why did you mention
+> Parallelism? And they are the same thing, right?
+
+Actually, _concurrency_ and _parallelism_ are not as the same. Let's dive into it a
+little bit.
+
+Roby pike has given a brilliant [talk](https://www.youtube.com/watch?v=cN_DpYBzKso) about it. In that, he
```
Comment (LeoSL) [inline _posts/2016-06-09-yet_another_introduction_to_concurrency_part_i.md]: Corta "in that"

**Pull request merged**